File: Plugins/Aspose_Imaging_Java_for_Ruby/lib/asposeimagingjava/metafiles/convertmetafiletootherformats.rb
module Asposeimagingjava
  module ConvertMetafileToOtherFormats
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load a Metafile in an instance of EmfMetafileImage class
        metafile = Rjb::import('com.aspose.imaging.fileformats.metafile.EmfMetafileImage').new(data_dir + "sample.emf")

        # Save EMF to BMP using BmpOptions object
        metafile.save(data_dir + "EmfToBmp.bmp", Rjb::import('com.aspose.imaging.imageoptions.BmpOptions').new)

        # Save EMF to JPG using JpegOptions object
        metafile.save(data_dir + "EmfToJpg.jpg", Rjb::import('com.aspose.imaging.imageoptions.JpegOptions').new)

        # Save EMF to PNG using PngOptions object
        metafile.save(data_dir + "EmfToPng.png", Rjb::import('com.aspose.imaging.imageoptions.PngOptions').new)

        # Save EMF to GIF using GifOptions object
        metafile.save(data_dir + "EmfToGif.gif", Rjb::import('com.aspose.imaging.imageoptions.GifOptions').new)

        # Save EMF to TIFF using TiffOptions object with default settings 
        metafile.save(data_dir + "EmfToTiff.tiff", Rjb::import('com.aspose.imaging.imageoptions.TiffOptions').new(Rjb::import('com.aspose.imaging.fileformats.tiff.enums.TiffExpectedFormat').Default))

        # Display Status.
        puts "Converted EMF to BMP, JPEG, PNG, GIF and TIFF formats successfully!"
    end
  end
end

File: Plugins/Aspose_Imaging_Java_for_PHP/src/aspose/imaging/ConvertingMetafilestoOtherImageFormats/ConvertMetafileToOtherFormats.php
<?php
namespace Aspose\Imaging\ConvertingMetafilestoOtherImageFormats;

use com\aspose\imaging\fileformats\metafile\EmfMetafileImage as EmfMetafileImage;
use com\aspose\imaging\imageoptions\BmpOptions as BmpOptions;
use com\aspose\imaging\imageoptions\JpegOptions as JpegOptions;
use com\aspose\imaging\imageoptions\PngOptions as PngOptions;
use com\aspose\imaging\imageoptions\GifOptions as GifOptions;
use com\aspose\imaging\imageoptions\TiffOptions as TiffOptions;
use com\aspose\imaging\fileformats\tiff\enums\TiffExpectedFormat as TiffExpectedFormat;

class ConvertMetafileToOtherFormats{

    public static function run($dataDir=null){

        # Load a Metafile in an instance of EmfMetafileImage class
        $metafile = new EmfMetafileImage($dataDir . "sample1.emf");

        # Save EMF to BMP using BmpOptions object
        $metafile->save($dataDir . "EmfToBmp.bmp", new BmpOptions());

        # Save EMF to JPG using JpegOptions object
        $metafile->save($dataDir . "EmfToJpg.jpg", new JpegOptions());

        # Save EMF to PNG using PngOptions object
        $metafile->save($dataDir . "EmfToPng.png", new PngOptions());

        # Save EMF to GIF using GifOptions object
        $metafile->save($dataDir . "EmfToGif.gif", new GifOptions());

        # Save EMF to TIFF using TiffOptions object with default settings
        $tiffExpectedFormat=new TiffExpectedFormat();
        $metafile->save($dataDir . "EmfToTiff.tiff", new TiffOptions($tiffExpectedFormat->Default));

        # Display Status.
        print "Converted EMF to BMP, JPEG, PNG, GIF and TIFF formats successfully!".PHP_EOL;
    }

}

?>